src/components/MultiHopTrade/MultiHopTrade.tsx

```diff
@@ -62,26 +62,57 @@ export const MultiHopTrade = memo(
       exact: true,
     })
 
-    const { chainId, assetSubId, sellChainId, sellAssetSubId, sellAmountCryptoBaseUnit } =
-      match?.params || {}
+    const {
+      chainId,
+      assetSubId,
+      sellChainId,
+      sellAssetSubId,
+      sellAmountCryptoBaseUnit: paramsSellAmountCryptoBaseUnit,
+    } = match?.params || {}
 
     const sellAsset = useAppSelector(selectInputSellAsset)
     const buyAsset = useAppSelector(selectInputBuyAsset)
     const history = useHistory()
     const sellInputAmountCryptoBaseUnit = useAppSelector(selectInputSellAmountCryptoBaseUnit)
     const [isInitialized, setIsInitialized] = useState(false)
+    const [isInitialMount, setIsInitialMount] = useState(true)
```

Comment:
Moved to a state variable. Ref updates immediately, and inherently break the rerender model. So after clearing the trade input slice, even though this is turbo fast (think immediate), it will take *at least* one more render for subscribers to be notified, vs. ref being instant as they're references, not memoized variables.

It being a ref was one of the main reasons this bug was happening, as clear would run, and next render the effect below this one (with the logic in this diff) would run, see `isInitialMount` to false, and rewrite URL with old sell amount (not yet cleared) and then bye bye